Build failure: input-leap

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```nix
  environment.systemPackages = (with pkgs; [
    input-leap
  ]);
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/294992978

### Relevant log output

```console
error: builder for '/nix/store/q67fr1gy6rs5h1q3rzjh1qxqggb41mpw-input-leap-3.0.2.drv' failed with exit code 2;
       last 25 log lines:
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:331:27: note: candidate: 'template<class T, typename std::enable_if<std::disjunction<std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, _Float16>, std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, qfloat16>, std::is_floating_point<_Tp> >::value, bool>::type <anonymous> > QString QString::arg(T, int, char, int, QChar) const'
       >   331 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, char format = 'g', int precision = -1,
       >       |                           ^~~
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:331:27: note:   template argument deduction/substitution failed:
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:330:50: error: no type named 'type' in 'struct std::enable_if<false, bool>'
       >   330 |     template <typename T, if_floating_point<T> = true>
       >       |                                                  ^~~~
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:336:27: note: candidate: 'template<class T, typename std::enable_if<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> >::value, bool>::type <anonymous> > QString QString::arg(const T&, int, QChar) const'
       >   336 |     [[nodiscard]] QString arg(const T &a, int fieldWidth = 0, QChar fillChar = u' ') const
       >       |                           ^~~
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:336:27: note:   candidate expects 3 arguments, 4 provided
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:356:5: note: candidate: 'template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const'
       >   356 |     arg(Args &&...args) const
       >       |     ^~~
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:356:5: note:   template argument deduction/substitution failed:
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h: In substitution of 'template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const [with Args = {char16_t&, int, int, QChar}]':
       > /build/source/src/gui/src/KeySequence.cpp:236:36:   required from here
       >   236 |         return QString("\\u%1").arg(QChar(key).toLower().unicode(), 4, 16, QChar('0'));
       >       |                ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qstring.h:356:5: error: no type named 'type' in 'struct std::enable_if<false, QString>'
       >   356 |     arg(Args &&...args) const
       >       |     ^~~
       > make[2]: *** [src/gui/CMakeFiles/guiunittests.dir/build.make:163: src/gui/CMakeFiles/guiunittests.dir/src/KeySequence.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:1156: src/gui/CMakeFiles/guiunittests.dir/all] Error 2
       > make: *** [Makefile:166: all] Error 2
       For full logs, run:
         nix log /nix/store/q67fr1gy6rs5h1q3rzjh1qxqggb41mpw-input-leap-3.0.2.drv
error: 1 dependencies of derivation '/nix/store/nzdmxhfvv0zrlng3r8ziqaag0khz5z38-system-path.drv' failed to build
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.2, NixOS, 25.05 (Warbler), 25.05.20250412.2631b0b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/5ds20jm3x2s4z7wn3581r6lc9ybmh45b-source`

### Notify maintainers

@KoviRobi, @phryneas, @Twey, @shymega

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/input-leap/default.nix
+++ b/pkgs/applications/misc/input-leap/default.nix
@@ -2,7 +2,6 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchpatch2,
   cmake,
 
   withLibei ? true,
@@ -29,13 +28,13 @@
 
 stdenv.mkDerivation rec {
   pname = "input-leap";
-  version = "3.0.2";
+  version = "3.0.3";
 
   src = fetchFromGitHub {
     owner = "input-leap";
     repo = "input-leap";
     rev = "v${version}";
-    hash = "sha256-YkBHvwN573qqQWe/p0n4C2NlyNQHSZNz2jyMKGPITF4=";
+    hash = "sha256-zSaeeMlhpWIX3y4OmZ7eHXCu1HPP7NU5HFkME/JZjuQ=";
     fetchSubmodules = true;
   };
 
@@ -65,14 +64,6 @@ stdenv.mkDerivation rec {
       libportal
     ];
 
-  patches = [
-    (fetchpatch2 {
-      # Upstream fix for crash on qt6.8 https://github.com/input-leap/input-leap/issues/2067
-      url = "https://github.com/input-leap/input-leap/commit/2641bc502e16b1fb7372b43e94d4b894cbc71279.patch?full_index=1";
-      hash = "sha256-LV09ITcE0ihKMByM5wiRetGwKbPrJbVY6HjZLqa8Dcs=";
-    })
-  ];
-
   cmakeFlags = [
     "-DINPUTLEAP_REVISION=${builtins.substring 0 8 src.rev}"
   ] ++ lib.optional withLibei "-DINPUTLEAP_BUILD_LIBEI=ON";
@@ -83,11 +74,6 @@ stdenv.mkDerivation rec {
       "''${gappsWrapperArgs[@]}"
         --prefix PATH : "${lib.makeBinPath [ openssl ]}"
     )
-  '';
-
-  postFixup = ''
-    substituteInPlace $out/share/applications/io.github.input_leap.InputLeap.desktop \
-      --replace "Exec=input-leap" "Exec=$out/bin/input-leap"
   '';
 
   meta = {

